src/extensions/mockFunctions.ts

```diff
@@ -0,0 +1,44 @@
+import {
+    BrsType,
+    BrsString,
+    ValueKind,
+    Callable,
+    StdlibArgument,
+    BrsInvalid,
+    RoAssociativeArray,
+} from "../brsTypes";
+import { Interpreter } from "../interpreter";
+
+export const mockFunctions = new Callable("mockFunctions", {
+    signature: {
+        args: [
+            new StdlibArgument("subtype", ValueKind.String),
+            new StdlibArgument("mock", ValueKind.Object),
+        ],
+        returns: ValueKind.Void,
+    },
+    impl: (interpreter: Interpreter, subtype: BrsString, mockFunctions: RoAssociativeArray) => {
+        let maybeComponent = interpreter.environment.nodeDefMap.get(subtype.value);
+
+        if (!maybeComponent) {
+            interpreter.stderr.write(
+                `Unable to mock functions for unknown roSGNode subtype ${subtype.value}`
+            );
+            return BrsInvalid.Instance;
+        }
+        if (mockFunctions instanceof RoAssociativeArray) {
+            [...mockFunctions.elements]
+                .filter(([_, value]) => value.kind === ValueKind.Callable)
+                .forEach(([key, value]) => {
+                    if (value instanceof Callable) {
```

Comment:
Do we just need this `if` statement to make the compiler happy? I'm curious how we would run into a scenario where we'd have `value.kind === ValueKind.Callable` but not `value instanceof Callable`

Reply:
Right. I mean if we are already filtering for `ValueKind.Callable` then we should not need an `if` check for `Callable` in the next expression. I think I added that to satisfy the compiler as you said.

Reply:
Yep, pretty sure that's a compiler affordance, but I suspect there's a better way around it :)